heed/src/databases/database.rs

```diff
@@ -2646,6 +2646,74 @@ impl<KC, DC, C, CDUP> Database<KC, DC, C, CDUP> {
         }
     }
 
+    /// Removes this database entirely.
+    /// The database is removed once the txn is committed.
+    ///
+    /// # Safety
+    ///
+    /// Ensure that no other copies of the database exist before calling, as
+    /// they will become invalid.
+    /// Do not remove a database if an existing transaction has modified it.
+    /// Doing so can cause database corruption or other errors.
+    ///
+    /// ```
+    /// # use std::fs;
+    /// # use std::path::Path;
+    /// # use heed::EnvOpenOptions;
+    /// use heed::Database;
+    /// use heed::types::*;
+    /// use heed::byteorder::BigEndian;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let dir = tempfile::tempdir()?;
+    /// # let env = unsafe { EnvOpenOptions::new()
+    /// #     .map_size(10 * 1024 * 1024) // 10MB
+    /// #     .max_dbs(3000)
+    /// #     .open(dir.path())?
+    /// # };
+    /// /// List databases in an env
+    /// fn list_dbs(env: &heed::Env) -> heed::Result<Vec<String>> {
+    ///     let rotxn = env.read_txn()?;
+    ///     let names_db: Database<Str, DecodeIgnore> =
+    ///         env.open_database(&rotxn, None)?
+    ///            .expect("the unnamed database always exists");
+    ///     let mut names = Vec::new();
+    ///     for item in names_db.iter(&rotxn)? {
+    ///         let (name, ()) = item?;
+    ///         names.push(name.to_owned());
+    ///     }
+    ///     Ok(names)
+    /// }
+    ///
+    /// type BEI32 = I32<BigEndian>;
+    ///
+    /// let mut rwtxn = env.write_txn()?;
+    /// let db: Database<BEI32, Str> = env.create_database(&mut rwtxn, Some("iter-i32"))?;
+    /// rwtxn.commit()?;
+    ///
+    /// let db_names = list_dbs(&env)?;
+    /// assert_eq!(db_names, vec!["iter-i32".to_owned()]);
+    ///
+    /// let mut rwtxn = env.write_txn()?;
+    /// unsafe { db.remove(&mut rwtxn)? };
+    ///
+    /// // DB still exists until the txn is committed
+    /// let db_names = list_dbs(&env)?;
+    /// assert_eq!(db_names, vec!["iter-i32".to_owned()]);
+    ///
+    /// rwtxn.commit()?;
+    /// let db_names = list_dbs(&env)?;
+    /// assert!(db_names.is_empty());
```

Comment:
Would you mind changing this example a bit so that the read txn is created in the main function and not inside, please?

Reply:
Which read txn are you referring to?

Reply:
Sorry, the one you open **inside** of `list_dbs`.

Reply:
Done! This caused me to catch an error in my docs - the DB is deleted immediately in the rwtxn, so my comment about DB delete taking effect only after committing a rwtxn was incorrect. I have removed the comment.